File: app/src/main/java/com/student/rentals/ui/fragments/home_fragment/HomeFragment.java
package com.student.rentals.ui.fragments.home_fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.student.models.ApartmentData;
import com.student.models.pApartmentData;
import com.student.rentals.R;
import com.student.rentals.ui.adapters.HouseListAdapter;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class HomeFragment extends Fragment {
    private static final String TAG = "HomeFragment";
    private View view;
    @BindView(R.id.house_list)
    RecyclerView house_list;
    private HomeViewModel viewModel;
    private HouseListAdapter adapter;

    public HomeFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.home_fragment, container, false);
        adapter = new HouseListAdapter(getActivity());
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        viewModel = new ViewModelProvider(this).get(HomeViewModel.class);
        viewModel.getRoomsList().observe(getViewLifecycleOwner(), new Observer<pApartmentData>() {
            @Override
            public void onChanged(pApartmentData pApartmentData) {
                createList(pApartmentData.getData());
            }
        });
    }

    @Override
    public void onActivityCreated(@Nullable final Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);


    }

    private void createList(List<ApartmentData> apartmentData) {
        house_list.setLayoutManager(new LinearLayoutManager(getContext()));
        adapter.addItems(apartmentData);
        house_list.setAdapter(adapter);
    }

}

File: app/src/main/java/com/student/rentals/ui/fragments/home_fragment/HomeFragment.kt
package com.student.rentals.ui.fragments.home_fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.student.models.DataApartment
import com.student.rentals.R
import com.student.rentals.ui.adapters.HouseListAdapter
import com.student.rentals.ui.fragments.BaseFragment
import kotlinx.android.synthetic.main.home_fragment.*
import kotlinx.android.synthetic.main.list_fragment.p_progress_bar

class HomeFragment : BaseFragment() {
    private val viewModel: HomeViewModel? by activityViewModels()
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.home_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel!!.roomsList.observe(viewLifecycleOwner, Observer { (_, _, data) -> createList(data) })
    }

    private fun createList(apartmentData: List<DataApartment>?) {
        house_list.layoutManager = LinearLayoutManager(context) as RecyclerView.LayoutManager?
        house_list.isNestedScrollingEnabled = false
        house_list.setHasFixedSize(true)
        house_list.setItemViewCacheSize(20)
        house_list!!.adapter = HouseListAdapter(activity, apartmentData)
        p_progress_bar.visibility = View.INVISIBLE
        house_list!!.visibility = View.VISIBLE
    }

    companion object {
        private const val TAG = "HomeFragment"
    }
}